src/bandersnatch/mirror.py

```diff
@@ -128,18 +128,17 @@ def _cleanup(self):
 
     def _filter_packages(self):
         """
-        Run the package filtering plugins and remove any packages from the
-        packages_to_sync that match any filters.
+        Run the package filtering plugins and remove any packages from
+        self.packages_to_sync that match any filters.
         """
         packages = list(self.packages_to_sync.keys())
```

Comment:
Why do we need to make this a list? Can we just use the generator for the loop below?
- The deletion happens on the key in the dict, so this is not needed from what I see.

Reply:
You cannot remove keys from a dictionary while iterating over them, the following code will generates an exception:
``` python
packages = dict(a=1,b=2,c=3)
packages.keys()
for package in packages:
    del packages[package]
packages.keys()
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
<ipython-input-1-b76a4ea0ad81> in <module>()
      1 packages = dict(a=1,b=2,c=3)
      2 packages.keys()
----> 3 for package in packages:
      4     del packages[package]
      5 packages.keys()

RuntimeError: dictionary changed size during iteration
```